### PR TITLE
expression: implement vectorized evaluation for `builtinCeilIntToDecSig`

### PR DESCRIPTION
**Pull request by hey-kong**
### What problem does this PR solve?

Implement vectorized evaluation for builtinCeilIntToDecSig.#12102

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinMathFunc/builtinCeilIntToDecSig-VecBuiltinFunc-4         59418             20651 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinMathFunc/builtinCeilIntToDecSig-NonVecBuiltinFunc-4      16982             67640 ns/op           39264 B/op        818 allocs/op
```

### Check List

Tests
- Unit test

### CODE REVIEW

File: expression/builtin_math_vec.go [hunk 1]
```diff
@@ -695,11 +696,36 @@ func (b *builtinRandWithSeedSig) vecEvalReal(input *chunk.Chunk, result *chunk.C
...
+
+	i64s := buf.Int64s()
+	d := result.Decimals()
+	isUnsigned := mysql.HasUnsignedFlag(b.args[0].GetType().Flag)
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if isUnsigned || i64s[i] >= 0 {
+			d[i] = *types.NewDecFromUint(uint64(i64s[i]))
```
Comment (qw4990) [inline expression/builtin_math_vec.go]: We should add a `continue` statement below this line according to the old implementation.

Comment (hey-kong) [reply expression/builtin_math_vec.go]: oh, I was careless.

**Pull request merged**